Facets XML (#311)

nikcio opened the pull request:
This adds XML documentation to the facets feature as well as other places in the project. This means that this PR is a super set of #311 and should be merged after the facets feature.

To see the changes in this PR before merging #311 see [this commit](https://github.com/Shazwazza/Examine/commit/5ab8b4dff0a3449e5358afaf36cd890f5d063d64#diff-15152175bf24ccd6097049544e6719125d556beeeb9c42c61ac2b363498b7173)

## What changed

- Added `<GenerateDocumentationFile>True</GenerateDocumentationFile>` to project files
- Fixed XML warnings
- Added XML to new facets feature.

## Approach

- The XML information has been added with the information that could be interpreted by looking in the source code and documentation.
- Added `<inheritdoc />` to constructors without XML information. Constructor XML information is rarely needed so adding `<inheritdoc />` removes the XML warning.

## Remaining XML warnings

Some classes have not had XML added due to not knowing what the class / method is used to do.

 - `GenericDirectoryFactory`
 - `FileSystemDirectoryFactory`
 - `IApplicationIdentifier`
 - `IApplicationIdentifier.GetApplicationUniqueIdentifier()`
 - `IApplicationRoot`
 - `IApplicationRoot.ApplicationRoot`

nzdev commented:
https://github.com/Shazwazza/Examine/pull/322 Should help with showing all the api xml documentation added in this pr on the docs site.

nikcio commented:
@Shazwazza I've updated this PR so it's now again ready to be merged. I also fixed 2 things that were missed in #311 :

- `LuceneFacetExtractionContext` now has a namespace
- `IFacetResult` now has the `TryGetFacet` method.

Shazwazza commented on src/Examine.Lucene/ReaderStatus.cs:
```diff
@@ -1,9 +1,27 @@
 {
-    public enum ReaderStatus { Current, Closed, NotCurrent }
+    /// <summary>
+    /// TODO: Is this enum used? (Found 0 references)
```
Looks like this is old, can't remember what from

nikcio replied:
@Shazwazza Should we delete it then or is it best to keep?

Shazwazza replied:
We should prob delete it

Shazwazza reviewed (approved):
Amazing!
